### ISSUE TITLE
Permission denied: Github releases deployment

### ISSUE THREAD

**Issue by chrisnharvey** [travis-ci/travis-ci#6397]
Hi,

I'm using travis-ci to deploy the eloquentos/packages build ourput to Github releases, but I can't seem it get it working as I keep getting the following error(s).

I read somewhere that tags containing only numeric characters may not work, but I've tried it with alpha-numeric characters and the error is the same. You can see my build output here https://travis-ci.org/eloquentos/packages/builds/148923092

This is a duplicate of travis-ci/dpl#488 as I'm not sure which repository this should be filed under. Please let me know which one should be closed.

Any help would be greatly appreciated

```
error: could not lock config file .git/config: Permission denied
error: could not lock config file .git/config: Permission denied
No stash found.
/usr/local/rvm/rubies/ruby-1.9.3-p551/lib/ruby/1.9.1/fileutils.rb:247:in `mkdir': Permission denied - .dpl (Errno::EACCES)
	from /usr/local/rvm/rubies/ruby-1.9.3-p551/lib/ruby/1.9.1/fileutils.rb:247:in `fu_mkdir'
	from /usr/local/rvm/rubies/ruby-1.9.3-p551/lib/ruby/1.9.1/fileutils.rb:221:in `block (2 levels) in mkdir_p'
	from /usr/local/rvm/rubies/ruby-1.9.3-p551/lib/ruby/1.9.1/fileutils.rb:219:in `reverse_each'
	from /usr/local/rvm/rubies/ruby-1.9.3-p551/lib/ruby/1.9.1/fileutils.rb:219:in `block in mkdir_p'
	from /usr/local/rvm/rubies/ruby-1.9.3-p551/lib/ruby/1.9.1/fileutils.rb:205:in `each'
	from /usr/local/rvm/rubies/ruby-1.9.3-p551/lib/ruby/1.9.1/fileutils.rb:205:in `mkdir_p'
	from /usr/local/rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.17/lib/dpl/provider.rb:128:in `deploy'
	from /usr/local/rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.17/lib/dpl/cli.rb:32:in `run'
	from /usr/local/rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.17/lib/dpl/cli.rb:7:in `run'
	from /usr/local/rvm/gems/ruby-1.9.3-p551/gems/dpl-1.8.17/bin/dpl:5:in `<top (required)>'
	from /usr/local/rvm/gems/ruby-1.9.3-p551/bin/dpl:23:in `load'
	from /usr/local/rvm/gems/ruby-1.9.3-p551/bin/dpl:23:in `<main>'
failed to deploy
```

**Comment by BanzaiMan** [comment]
You are changing ownership of the build directory https://travis-ci.org/eloquentos/packages/builds/148923092#L236:

```sh-session
$ sudo chown 1000:1000 -R ${PWD}
```

I don't know this command's goal, but I suspect it is incompatible with the deployment mechanism, which assumes that it can manipulate files at will as user `travis`. https://github.com/travis-ci/dpl/blob/v1.8.17/lib/dpl/provider.rb#L128. (The warnings regarding `.git/config` are coming from https://github.com/travis-ci/dpl/blob/v1.8.17/lib/dpl/provider.rb#L193-L195.)

**Comment by chrisnharvey** [comment]
Hi @BanzaiMan 

Thanks for your reply.

Yeah, this was required for my script. I was changing the owner back to travis in the ```after_script``` section, which I've only just realised runs *after* deployment.

I have now changed ```after_script``` to ```after_success```. So hopefully the owner will be changed before deployment which should hopefully fix my issue.

It is currently building now, so I'll update this issue if its successful or not.

https://travis-ci.org/eloquentos/packages/builds/148943781
